osu.Game.Rulesets.Osu.Tests/SpinnerSpinHistoryTest.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using NUnit.Framework;
+using osu.Game.Rulesets.Osu.Objects.Drawables;
+
+namespace osu.Game.Rulesets.Osu.Tests
+{
+    [TestFixture]
+    public class SpinnerSpinHistoryTest
```

Comment:
Hmm, I get that this is intending to have test coverage of various rewind scenarios, but it's quite hard to follow with this level of abstraction. Wondering if we can make `TestSpinnerInput` also cover some rewind cases...

Reply:
I've also added a visual test to that one.